javascript/stimulus_reflex.js

```diff
@@ -1,48 +1,92 @@
-  const { channel, room } = controller.StimulusReflex;
-  const id = `${channel}${room}`;
-  const renderDelay = controller.StimulusReflex.renderDelay || 25;
-
+  const { channel, room } = controller.StimulusReflex
+  const id = `${channel}${room}`
+  const renderDelay = controller.StimulusReflex.renderDelay || 25
   const subscription =
     app.StimulusReflex.subscriptions[id] ||
     app.StimulusReflex.consumer.subscriptions.create(
       { channel, room },
       {
         received: data => {
           if (data.cableReady) {
-            clearTimeout(controller.StimulusReflex.timeout);
+            clearTimeout(controller.StimulusReflex.timeout)
             controller.StimulusReflex.timeout = setTimeout(() => {
-              CableReady.perform(data.operations);
-            }, renderDelay);
+              CableReady.perform(data.operations)
+            }, renderDelay)
           }
-        },
+        }
       }
-    );
+    )
 
-  app.StimulusReflex.subscriptions[id] = subscription;
-  controller.StimulusReflex.subscription = subscription;
-};
+  app.StimulusReflex.subscriptions[id] = subscription
+  controller.StimulusReflex.subscription = subscription
+}
 
 const extend = controller => {
+  const url = location.href
   Object.assign(controller, {
-    stimulate() {
-      clearTimeout(controller.StimulusReflex.timeout);
-      const url = location.href;
-      let args = Array.prototype.slice.call(arguments);
-      let target = args.shift();
-      controller.StimulusReflex.subscription.send({ target, args, url });
+    stimulate () {
+      clearTimeout(controller.StimulusReflex.timeout)
+      const args = Array.prototype.slice.call(arguments)
+      const target = args.shift()
+      controller.StimulusReflex.subscription.send({ target, args, url })
+    },
+    reflex (el) {
+      clearTimeout(controller.StimulusReflex.timeout)
+      const segments = el.target.dataset.reflex.split('->')
+      const name =
+        segments.length == 1 ? segments[0].split('#') : segments[1].split('#')
+      const target = `${name[0].charAt(0).toUpperCase() +
+        name[0].slice(1)}Reflex#${name[1]}`
+      let args = []
+      for (const arg in el.target.dataset) {
+        if (/^reflexValue/.test(arg)) args.push(el.target.value)
+        else if (/^reflexChecked/.test(arg)) {
+          args.push(el.target.checked ? '1' : '0') 
+        } else if (/^reflex.+/.test(arg)) args.push(el.target.dataset[arg])
+      }
+      controller.StimulusReflex.subscription.send({ target, args, url })
     },
-  });
-};
+    wire (e) {
+      const method =
+        e && e.type === 'cable-ready:before-morph'
+          ? 'removeEventListener'
+          : 'addEventListener'
+      const events = {
+        a: 'click',
+        button: 'click',
+        form: 'submit',
+        input: 'change',
+        select: 'change',
+        textarea: 'change'
+      }
+      document.querySelectorAll('[data-reflex]').forEach(el => {
```

Comment:
This 100% needs to be scoped to the DOM element that the controller is declared on, or else you'll create duplicate events.